app/workers/data_request_worker.rb

```diff
@@ -12,37 +12,59 @@ def perform(request_id)
     begin
       request = DataRequest.find(request_id)
       return if request.processing? || request.complete?
+      request.processing!
 
       self.path = Rails.root.join("tmp", "#{request.id}.csv")
 
-      request.processing!
-
-      exporter = if request.extracts?
-        Exporters::CsvExtractExporter
-      elsif request.reductions?
-        Exporters::CsvSubjectReductionExporter
-      end.new(
-        :resource_id => request.exportable.id,
-        :resource_type => request.exportable.class,
-        :user_id => request.user_id,
-        :subgroup => request.subgroup
+      exporter = Exporters::CsvExporter.new(
+        resource_id: request.exportable.id,
+        resource_type: request.exportable.class.name,
+        user_id: request.user_id,
+        subgroup: request.subgroup,
+        requested_data: request.requested_data
       )
 
-      exporter.dump(path) do |progress, total|
+      related_counter = "#{request.requested_data}_count"
```

Comment:
Is adding counters here purely because you're already iterating row by row anyway, might as well count em and update the cache?

Reply:
Yeah that's exactly it! We'll provide the cached value as an estimate and then when we finish we'll know the exact value, so we may as well use it.